app/controllers/tweets_controller.rb

```diff
@@ -0,0 +1,77 @@
+class TweetsController < ApplicationController
+
+    get '/tweets' do
+        if logged_in?
+            @user = current_user
+            @tweets = Tweet.all
+            erb :"tweets/index"
+        else
+            redirect "/login"
+        end
+    end 
+
+    get '/tweets/new' do
+        if logged_in?
+            @user = current_user
+            erb :"tweets/new"
+        else
+            redirect "/login"
+        end
+    end
+
+    post '/tweets' do
+        if logged_in? && !params[:content].empty?
+            @user = current_user
+            @user.tweets.create(content:params[:content])
+            redirect "/users/#{@user.slug}"
+        elsif logged_in? && params[:content].empty?
+            redirect "/tweets/new"
+        else
+            redirect "/login"
+        end
+    end
+
+    patch '/tweets/:id' do
+        if logged_in? && Tweet.find(params[:id]).user == current_user && !params[:content].empty?
+            Tweet.find(params[:id]).update(content:params[:content])
+            redirect "/users/#{current_user.slug}"
+        elsif logged_in?
+            redirect "/tweets/#{params[:id]}/edit"
+        else
+            redirect "/login"
+        end
+    end
+
+    get '/tweets/:id' do
+        if logged_in?
+            @tweet = Tweet.find(params[:id])
+            erb :"tweets/show"
+        else
+            redirect "/login"
+        end
+    end
+
+    get '/tweets/:id/edit' do
+        if logged_in? && Tweet.find(params[:id]).user == current_user
+            @tweet = Tweet.find(params[:id])
+            erb :"tweets/edit"
+        elsif logged_in? && Tweet.find(params[:id]).user != current_user
+            redirect "/tweets"
+        else
+            redirect "/login"
+        end
+    end
+    
+    delete '/tweets/:id/delete' do
+        if logged_in? && Tweet.find(params[:id]).user == current_user
```

Comment:
Prefer to find tweets that are already only associated to the `current_user` than to manually check the `user_id` foreign key of the tweet against the `current_user.id`

```ruby
tweet = current_user.tweets.find_by(id: params[:id])
if tweet && tweet.destroy
  redirect "/tweets"
else
  redirect "/tweets/#{tweet.id}"
end
```

We're using the association collection method provided by [`has_many`](http://api.rubyonrails.org/classes/ActiveRecord/Associations/ClassMethods.html#method-i-has_many) to query the user's tweets. 

Note the usage of [`find_by`](http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find-by) vs [`find`](http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find). Had we used `find` here, ActiveRecord would raise an error in the event that we could not find a tweet associated with this user. `find_by` will allow this action to fail gracefully, simply returning nil if the user is trying to delete a tweet that they do not have access to.